webapp/src/ts/services/check-date.service.ts

```diff
@@ -0,0 +1,61 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { TelemetryService } from './telemetry.service';
+import { ModalService } from '@mm-modals/mm-modal/mm-modal';
+import { CheckDateComponent } from '@mm-modals/check-date/check-date.component';
+
+const A_DATE_IN_THE_PAST = 1454424982000;
+const MARGIN_OF_ERROR = 10 * 60 * 1000; // ten minutes
+
+interface CheckDateData {
+  reportedLocalDate: Date;
+  expectedLocalDate?: Date;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CheckDateService {
+
+  constructor(
+    private http: HttpClient,
+    private telemetryService: TelemetryService,
+    private modalService: ModalService,
+  ) {
+  }
+
+  showModal(dates: CheckDateData) {
+    this.modalService.show(CheckDateComponent, {initialState: dates});
+  }
+
+  check() {
+    return this.http.head('/api/info?seed=' + Math.random(), { observe: 'response' })
```

Comment:
Please put `head` on a new line. Please take the time and read our style guide: https://docs.communityhealthtoolkit.org/contribute/code/style-guide/#method-chaining

Reply:
Yes sorry, definitely this is the case for the chaining style because the line is kind of long, but it is not a mandatory rule ("..._if you want to chain methods..._"), maybe we should set a max number of chars allowed without the chaining indentation so is easy to follow. Now a max of 120 chars is enforced to any line, maybe we should enforce chaining indentation if the line has more than 80 chars, or 60?...

Reply:
Reading again the statement "...if you want to chain methods..." yes it mean we always should use indentation (though _should_ is not _must_) but still I think it is too much for certain expression, eg.:

```js
User
  .find({ name: 'foo' })
  .toPromise();
```

Reply:
I agree that when  chaining 2 methods, and they fit one one line, it's fine to inline them: `foo.bar().baz();`, when there's 3 ore more, definitely they should go on the next line, and definitely no mixed chaining:
```
User.findOne({ name: 'foo' })
  .populate('bar')
  .exec(() => true);
```